zebra-network/src/peer_set/initialize.rs

```diff
@@ -194,9 +194,10 @@ where
     S::Future: Send + 'static,
 {
     info!(?initial_peers, "Connecting to initial peer set");
-    use tower::util::CallAllUnordered;
-    let addr_stream = futures::stream::iter(initial_peers.into_iter());
-    let mut handshakes = CallAllUnordered::new(connector, addr_stream);
+    let mut handshakes = initial_peers
+        .iter()
+        .map(|request| connector.clone().oneshot(*request))
+        .collect::<futures::stream::FuturesUnordered<_>>();
```

Comment:
We don't want to use `try_collect` here:
> If an error happens then all collected elements will be dropped and the error will be returned.

https://docs.rs/futures/0.3.12/futures/stream/trait.TryStreamExt.html#method.try_collect

Instead, we just pass the `FuturesUnordered` to the existing code, which expects a `CallAllUnordered` (they are similar types).

Reply:
I also removed a `clone` for you.

Reply:
Was the example code from PR #1620 helpful?
```rust
               hashes
                    .into_iter()
                    .map(|hash| zs::Request::Block(hash.into()))
                    .map(|request| self.state.clone().oneshot(request))
                    .collect::<futures::stream::FuturesOrdered<_>>()
```
https://github.com/ZcashFoundation/zebra/pull/1620/files#diff-3f14971a74a109913a4809bd1a77f4e0145e1edfa72aaded8f1fdcb2cc45f4f1L188